### PR TITLE
Added Relativisitc Average GAN Loss 

### PR DESCRIPTION
**Pull request by nivedwho**
Added two functions 
- ragan_generator() for calculating generator loss. 
- ragan_discriminator() for calculating discriminator loss

and a class RelativisticAverageLoss for testing the two functions.

### CODE REVIEW

File: tensorflow_gan/python/losses/losses_impl.py [hunk 1]
```diff
@@ -58,6 +58,8 @@
     'mutual_information_penalty',
     'combine_adversarial_loss',
     'cycle_consistency_loss',
+    'ragan_generator_loss',
```
Comment (joel-shor) [inline tensorflow_gan/python/losses/losses_impl.py:61]: what do you think about the names "relativistic_generator_loss" instead? I think it's more descriptive.

File: tensorflow_gan/python/losses/losses_impl.py [hunk 2]
```diff
@@ -1155,3 +1157,83 @@ def cycle_consistency_loss(data_x,
       tf.compat.v1.summary.scalar('cycle_consistency_loss', loss)
 
   return loss
+
+
+def ragan_discriminator_loss(d_real, 
+                             d_fake,
```
Comment (joel-shor) [inline tensorflow_gan/python/losses/losses_impl.py:1163]: Can you please use the same naming conventions as the other losses in this file? See any of the other discriminator loss functions for what to call these

File: tensorflow_gan/python/losses/losses_impl.py [hunk 3]
```diff
@@ -1155,3 +1157,83 @@ def cycle_consistency_loss(data_x,
...
+
+    real_loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(
+            labels=tf.ones_like(real_logits), logits=real_logits))
+    fake_loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(
+        labels=tf.zeros_like(fake_logits), logits=fake_logits))
+
+  return real_loss + fake_loss
+
+def ragan_generator_loss(d_real, 
+                         d_fake,
```
Comment (joel-shor) [inline tensorflow_gan/python/losses/losses_impl.py:1202]: same as above

**Review by joel-shor — CHANGES_REQUESTED** [review]
Name changes only. Other than that, looks great. Please ping me when this is completed!

**Review by joel-shor — CHANGES_REQUESTED** [review]
One-line docstring summary should end with ".", ".)", "?", or "!" [g-short-docstring-punctuation]
	//depot/google3/third_party/py/tensorflow_gan/python/losses/tuple_losses_test.py:272:1
One-line docstring summary should end with ".", ".)", "?", or "!" [g-short-docstring-punctuation]
	//depot/google3/third_party/py/tensorflow_gan/python/losses/tuple_losses_test.py:292:1

**Comment by joel-shor** [comment]
Responded!
